Cabal/Distribution/Parsec/Types/Common.hs

```diff
@@ -0,0 +1,64 @@
+-- | Module containing small types
+module Distribution.Parsec.Types.Common (
+    -- * Diagnostics
+    PError (..),
+    PWarning (..),
+    showPWarning,
+    -- * Field parser
+    FieldParser (..),
+    -- * Position
+    Position (..),
+    incPos,
+    retPos,
+    showPos,
+    ) where
+
+import Prelude ()
+import Distribution.Compat.Prelude
+
+import System.FilePath (normalise)
+
+import qualified Data.ByteString as BS
+import qualified Text.Parsec as Parsec
+
+data PError = PError Position String
+    deriving (Show)
+data PWarning = PWarning Position String
+    deriving (Show)
+
+ -- TODO: showPerror, remove Show instan
+ --
+showPWarning :: FilePath -> PWarning -> String
+showPWarning fpath (PWarning pos msg) =
+  normalise fpath ++ ": " ++ show pos ++ msg
+
+-------------------------------------------------------------------------------
+-- Field parser
+-------------------------------------------------------------------------------
+
+data FieldParser a
```

Comment:
Because for some fields it isn't necessary to decode utf8. At this point I'm not sure though how much non-utf8 fields we have, so might be that this becomes a type synonym.